python/Ganga/new_tests/GPI/Loading/LazyLoadingTest.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import absolute_import
+
+from ..GangaUnitTest import GangaUnitTest
+
+class LazyLoadingTest(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(LazyLoadingTest, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+
+    def test_a_JobConstruction(self):
+        """ First construct the Job object (singular)"""
+        from Ganga.Utility.Config import getConfig
+        self.assertFalse(getConfig('TestingFramework')['AutoCleanup'])
+
+        from Ganga.GPI import Job, jobs
+        j=Job()
+        self.assertTrue(len(jobs) == 1)
+
+    def test_b_JobNotLoaded(self):
+        """ Second get the job and check that getting it via jobs doesn't cause it to be loaded"""
+        from Ganga.GPI import jobs
+
+        self.assertTrue(len(jobs) == 1)
```

Comment:
We're still using 2.6 on Jenkins since for now we need to make sure that it works there. I am tempted however to make unittest2 a requirement for the testing so that we can use all the nice features in Python 2.7's `unittest` module

Currently all tests are running in series so they are all running on the same repo (`~/gangadir_testing/`). In the medium-term I'd like to investigate running every test on a separate repo (so `GPI/Loading/LazyLoadingTest.py` would run with `gangadir` as `~/gangadirs/GPI-Loading-LazyLoadingTest/` or something). This way it will be possible to run multiple tests in parallel and if one is a long running Grid job then it won't hold anything up.